internal/connect/wrapper.go

```diff
@@ -0,0 +1,89 @@
+package connect
+
+import (
+	"fmt"
+
+	"github.com/SUSE/connect-ng/internal/collectors"
+	"github.com/SUSE/connect-ng/internal/credentials"
+	"github.com/SUSE/connect-ng/pkg/connection"
+	"github.com/SUSE/connect-ng/pkg/registration"
+)
+
+// Wrapper is a bridge between API connections via `pkg/connection/` and
+// `internal/credentials/`. Use this wrapper in order to perform any API
+// requests on the context of SUSEConnect.
+type Wrapper struct {
+	// Connection is the API connection as defined in `pkg/connection/`.
+	Connection *connection.ApiConnection
+
+	// Whether the current system is registered or not. Set after calling `New.`
+	Registered bool
+}
+
+// Returns a new Wrapper object by taking the given Options into account. Note
+// that it will also make an attempt to read any available credentials, and set
+// Wrapper.Registered accordingly.
+func New(opts *Options) *Wrapper {
```

Comment:
Can we rename this to `NewWrapper` to make it more speaking in the rest of the code?